server/server.go

```diff
@@ -993,6 +993,8 @@ func (a *ArgoCDServer) newHTTPServer(ctx context.Context, port int, grpcWebHandl
 	}
 	if len(a.ContentTypes) > 0 {
 		handler = enforceContentTypes(handler, a.ContentTypes)
+	} else {
```

Comment:
@crenshaw-dev Is there a situation where someone could upgrade their manifests and the content types option is not set? I sort of think the if/else should be opposite handling, only when you expressly turn them off should we turn them off, other wise we default to them being on. You know what I mean?

Reply:
Confirmed, if the user does not affirmatively set some value, the default is secure. To turn off validation, they must affirmatively set the allow list to an empty string.